examples/smoke-co-alarm-app/silabs/src/SmokeCoAlarmManager.cpp

```diff
@@ -37,10 +37,29 @@ CHIP_ERROR SmokeCoAlarmManager::Init()
 bool SmokeCoAlarmManager::StartSelfTesting()
 {
     bool success = true;
+    ExpressedStateEnum expressedState;
 
-    SILABS_LOG("Start self-testing!");
+    success = SmokeCoAlarmServer::Instance().GetExpressedState(1, expressedState);
 
-    SILABS_LOG("End self-testing!");
+    if (success)
+    {
+        success = SmokeCoAlarmServer::Instance().SetExpressedState(1, ExpressedStateEnum::kTesting);
```

Comment:
Again: _this_ part should live in the cluster, because all applications will need to do it.

The app knows what its expressed state is (the attr store is not the source of truth for that), and once the self-test is done should update the expressed state to the right value.  But setting it to kTesting should happen in the cluster itself, I would think.

Reply:
I have moved this part to cluster.